utils/calendrical_calculations/src/julian.rs

```diff
@@ -21,78 +26,76 @@ pub const fn is_leap_year(year: i32) -> bool {
 
 /// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1689-L1709>
 pub const fn fixed_from_julian(year: i32, month: u8, day: u8) -> RataDie {
-    let mut fixed =
-        JULIAN_EPOCH.to_i64_date() - 1 + 365 * (year as i64 - 1) + (year as i64 - 1).div_euclid(4);
-    debug_assert!(month > 0 && month < 13, "Month should be in range 1..=12.");
-    fixed += match month {
-        1 => 0,
-        2 => 31,
-        3 => 59,
-        4 => 90,
-        5 => 120,
-        6 => 151,
-        7 => 181,
-        8 => 212,
-        9 => 243,
-        10 => 273,
-        11 => 304,
-        12 => 334,
-        _ => -1,
-    };
-    // Only add one if the month is after February (month > 2), since leap days are added to the end of February
-    if month > 2 && is_leap_year(year) {
-        fixed += 1;
-    }
-    RataDie::new(fixed + (day as i64))
+    day_before_year(year)
+        .add(days_before_month(year, month) as i64)
+        .add(day as i64)
 }
 
-/// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1711-L1738>
-pub fn julian_from_fixed(date: RataDie) -> Result<(i32, u8, u8), I32CastError> {
-    let approx = (4 * date.to_i64_date() + 1464).div_euclid(1461);
-    let year = i64_to_i32(approx)?;
-    let prior_days = date
-        - fixed_from_julian(year, 1, 1)
-        - if is_leap_year(year) && date > fixed_from_julian(year, 2, 28) {
-            1
-        } else {
+/// The number of days in this year before this month starts
+///
+/// Inspired by Neri-Schneider <https://onlinelibrary.wiley.com/doi/10.1002/spe.3172>
+pub const fn days_before_month(year: i32, month: u8) -> u16 {
```

Comment:
nit: would prefer tests for non-obvious function impls

(non-blocking for Julian since it's a second-tier calendar in my book)

Reply:
From a code readability (and maintainability) perspective it is nice to be able to be assured that unreadable code A has the same semantics as readable code B.

A _lot_ of my time debugging calendrical_calculations issues has been in strange unreadable code based off of the Lisp where I don't have a great intuition as to what's going on. I'd prefer those situations to not be made more likely.